pkg/operator/controllers/pullsecret/pullsecret_controller.go

```diff
@@ -68,19 +77,18 @@ func NewReconciler(log *logrus.Entry, client client.Client) *Reconciler {
 //     changes, we'll see the pull Secret object requested.
 func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
 	instance := &arov1alpha1.Cluster{}
-	err := r.client.Get(ctx, types.NamespacedName{Name: arov1alpha1.SingletonClusterName}, instance)
+	err := r.AROController.Client.Get(ctx, types.NamespacedName{Name: arov1alpha1.SingletonClusterName}, instance)
 	if err != nil {
 		return reconcile.Result{}, err
 	}
 
 	if !instance.Spec.OperatorFlags.GetSimpleBoolean(controllerEnabled) {
-		r.log.Debug("controller is disabled")
+		r.AROController.Log.Debug("controller is disabled")
 		return reconcile.Result{}, nil
 	}
 
-	r.log.Debug("running")
-	userSecret := &corev1.Secret{}
-	err = r.client.Get(ctx, pullSecretName, userSecret)
+	r.AROController.Log.Debug("running")
+	userSecret, err := r.secretsClient.Get(ctx, pullSecretName.Name, metav1.GetOptions{})
```

Comment:
It's confusing to me why we need `secretsClient` and why we aren't just using `r.AROController.Client`